sklearn/gaussian_process/_gpr.py

```diff
@@ -355,18 +354,21 @@ def predict(self, X, return_std=False, return_cov=False):
 
                 return y_mean, y_cov
             elif return_std:
-                # cache result of K_inv computation
+                # cache result of K_inv computation,
+                # this is done for compatibility, K_inv should be avoided
```

Comment:
We don't need `self._K_inv` indeed. It is a private variable. We can remove the `self._K_inv` from the `fit` as well. Skipping this computation will be beneficial in terms of computational performance.